.github/workflows/code-styling.yml

```
@@ -0,0 +1,75 @@
+name: Code styling
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 1 * *" # first of every month at midnight
+
+
+concurrency:
+  # only one run per branch at a time
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  style-code:
+    runs-on: ubuntu-latest
+    name: Perform code styling
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Load 1Password secrets
+        uses: 1password/load-secrets-action@v1
+        with:
+          export-env: true
+        env:
+          # TODO: Which tokens do we actually need here?
+          OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_DOCS_BOT_SERVICE_TOKEN }}
+          DOCS_BOT_GITHUB_TOKEN: ${{ secrets.OP_DOCS_BOT_GITHUB_TOKEN }}
+
+      ## First, style the R code ##
+      - name: Install R
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Install R dependencies
+        run: install.packages("styler", repos = "https://packagemanager.posit.co/cran/__linux__/jammy/latest")
+        shell: Rscript --no-init-file {0}
+
+      - name: Style analysis modules R code
+        run: styler::style_dir("analyses", filetype = c("R", "r", "Rmd", "rmd")) # exclude_dirs and exclude_files do not seem to like globs...
+        shell: Rscript --no-init-file {0}
+
+      # maybe - hack since it seems to want to restyle renv/activate.R which maybe we don't want?
+      #- name: Check out any renv files modified by the styler
+
+
+      ## Second, style the python code ##
+      - name: Install python
+        uses: actions/setup-python@v2
+
+      - name: Install python dependencies
+        run: pip install ruff
+
+      - name: Style analysis modules python code
+        run: ruff format analyses/
```

The [euff github page](https://github.com/astral-sh/ruff?tab=readme-ov-file) suggests just using this action:
```suggestion
      - name: Install python
        uses: chartboost/ruff-action@v1
        with:
          args: format analyses/
```

---

👍 I had come across this but did not realize it was _recommended_